src/config/session.c

```diff
@@ -33,6 +34,36 @@ static const char *const env_vars[] = {
 	NULL
 };
 
+static long
+get_num(const char *str)
+{
+	if (string_null_or_empty(str)) {
+		return -1;
+	}
+
+	char c, *end = NULL;
+	long val = strtol(str, &end, 0);
+	if (!val || val == LONG_MAX) {
+		return -1;
+	}
+	while ((c = *end++) != 0) {
+		switch (c) {
+		case 'k':
```

Comment:
lowercase `k` but uppercase `M` and `G` seems a bit weird to me. Also: do we actually need the string parsing in the first place? I think the limit could also be a simple define.